src/java.base/share/classes/java/util/HexDigits.java

```diff
@@ -85,11 +74,11 @@ final class HexDigits implements Digits {
         short[] digits = new short[16 * 16];
 
         for (int i = 0; i < 16; i++) {
-            short hi = (short) ((i < 10 ? i + '0' : i - 10 + 'a') << 8);
+            short lo = (short) (i < 10 ? i + '0' : i - 10 + 'a');
 
             for (int j = 0; j < 16; j++) {
-                short lo = (short) (j < 10 ? j + '0' : j - 10 + 'a');
-                digits[(i << 4) + j] = (short) (hi | lo);
+                short hi = (short) ((j < 10 ? j + '0' : j - 10 + 'a') << 8);
+                digits[(i << 4) + j] = (short) (lo | hi);
```

Comment:
The part that is counter-intuitive is that the bits that are shifted left appear to the right of the bits that are not shifted. (the actual bits are correct, either way, but it reads oddly). In a short, the resulting bits would be `hhll`.
```suggestion
                digits[(i << 4) + j] = (short) (hi | lo);
```
Similarly, the returns of the other `packDigits` methods do not read MSB to LSB.

Reply:
You are right, I have changed